src/Android/BluetoothHelper.cpp

```diff
@@ -67,10 +67,16 @@ BluetoothHelper::Initialise(JNIEnv *env)
   createServer_method = env->GetStaticMethodID(cls, "createServer",
                                                "()Lorg/xcsoar/AndroidPort;");
 
+#if __ANDROID_API__ >= 21
+
   startLeScan_method = env->GetStaticMethodID(cls, "startLeScan",
                                               "(Landroid/bluetooth/BluetoothAdapter$LeScanCallback;)Z");
   stopLeScan_method = env->GetStaticMethodID(cls, "stopLeScan",
                                              "(Landroid/bluetooth/BluetoothAdapter$LeScanCallback;)V");
+#else /* __ANDROID_API__ >= 21 */
+  startLeScan_method = nullptr;
```

Comment:
If you really want to make this a compile-time check (why?), why does the variable `startLeScan_method` even exist?
Second question: why do you want to make this a compile-time check? It never was, and it has always worked with even older Android versions.